source/sessions/driver-sessions.rst

```diff
@@ -1197,6 +1212,25 @@ Instead, we require users to pass session as a parameter to each function::
   collection.insertOne(session, {})
   session.endSession()
 
+Why does a network error cause the ``ServerSession`` to be discarded from the pool?
+-----------------------------------------------------------------------------------
+
+When a network error is encountered when executing an operation with a
+``ClientSession``, the operation may be left running on the server. Re-using
+this ``ServerSession`` can lead to parallel operations which violates the
+rule that a session must be used sequentially. This results in multiple
+problems:
+
+#. killSessions to end an earlier operation would surprisingly also end a
+   later operation.
```

Comment:
What is the risk in `killSessions` ending a later operation? Is this only when dealing with implicit sessions? If you're using explicit sessions, regardless of whether the session is dirty or not, I would expect the user to realize that kill the session could end any operation that has not received an acknowledged response from the server. Some examples (with implicit or explicit):

**Implicit session without dirty tracking.** I execute a long-running updateMany and the socket dies. The operation is still running. I execute a second updateMany, which uses the same implicit session, and it succeeds. I then kill the implicit session's LSID and only the first operation is aborted. Even if multiple long-running operations were executed within that session, only those that encountered socket errors (or were otherwise unacknowledged and might have been executing asynchronously) will be aborted. I don't see how this relates to "ending a later operation".

**Explicit session without dirty tracking.** Similarly, I would expect killSessions to possibly abort operations that I've executed but have yet to receive a response (the could still be running). I'm not sure what a "later operation" would be in this context, either.